kafka-admin/src/main/java/org/bf2/admin/security/TrustedSSOCallerPrincipalFactoryProducer.java

```diff
@@ -0,0 +1,126 @@
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ */
+package org.bf2.admin.security;
+
+import io.smallrye.jwt.auth.principal.DefaultJWTCallerPrincipal;
+import io.smallrye.jwt.auth.principal.DefaultJWTCallerPrincipalFactory;
+import io.smallrye.jwt.auth.principal.DefaultJWTTokenParser;
+import io.smallrye.jwt.auth.principal.JWTAuthContextInfo;
+import io.smallrye.jwt.auth.principal.JWTCallerPrincipal;
+import io.smallrye.jwt.auth.principal.JWTCallerPrincipalFactory;
+import io.smallrye.jwt.auth.principal.KeyLocationResolver;
+import io.smallrye.jwt.auth.principal.ParseException;
+import org.bf2.admin.kafka.admin.KafkaAdminConfigRetriever;
+import org.eclipse.microprofile.config.Config;
+import org.eclipse.microprofile.config.ConfigProvider;
+import org.jose4j.http.Get;
+import org.jose4j.jwk.HttpsJwks;
+import org.jose4j.jwt.consumer.JwtContext;
+import org.jose4j.keys.resolvers.VerificationKeyResolver;
+import org.jose4j.lang.UnresolvableKeyException;
+
+import javax.annotation.Priority;
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.Alternative;
+import javax.enterprise.inject.Produces;
+import javax.interceptor.Interceptor;
+
+import java.util.logging.Logger;
+
+/**
+ * CDI producer to provide instance of {@code JWTCallerPrincipalFactory} capable
+ * of using a trusted certificate for JWKS retrieval.
+ *
+ * @deprecated needed until https://github.com/smallrye/smallrye-jwt/issues/571
+ *             is fixed and released
+ */
+@Deprecated(forRemoval = true)
+@ApplicationScoped
+@Alternative
+@Priority(Interceptor.Priority.APPLICATION + 10)
+public class TrustedSSOCallerPrincipalFactoryProducer {
+
+    private static final Logger LOG = Logger.getLogger(TrustedSSOCallerPrincipalFactoryProducer.class.getName());
+
+    private JWTCallerPrincipalFactory factory;
+
+    public TrustedSSOCallerPrincipalFactoryProducer() {
+        initCallerPrincipalFactory();
+    }
+
+    @Produces
+    public JWTCallerPrincipalFactory getFactory() {
+        return factory;
+    }
+
+    private void initCallerPrincipalFactory() {
+        LOG.info("Loading JWTCallerPrincipalFactory from " + getClass());
+        factory = new TrustedSSOCallerPrincipalFactory();
+        JWTCallerPrincipalFactory.setInstance(factory);
+    }
+
+    private static class TrustedSSOCallerPrincipalFactory extends DefaultJWTCallerPrincipalFactory {
+        private final DefaultJWTTokenParser parser = new TrustedSSOTokenParser();
+
+        @Override
+        public JWTCallerPrincipal parse(final String token, final JWTAuthContextInfo authContextInfo) throws ParseException {
+            JwtContext jwtContext = parser.parse(token, authContextInfo);
+            String type = jwtContext.getJoseObjects().get(0).getHeader("typ");
```

Comment:
To my untrained eye, "typ" looks like a typo, but I presume it's not.

Reply:
I should have mentioned more detail, but all of this is meant to be able to override the `initializedHttpJwks` method so that the trusted cert can be set for the HTTP request. Otherwise this should be a copy of the upstream code calling an overridden `parser` field:
https://github.com/smallrye/smallrye-jwt/blob/main/implementation/jwt-auth/src/main/java/io/smallrye/jwt/auth/principal/DefaultJWTCallerPrincipalFactory.java

This is the upstream method that is being overridden:
https://github.com/smallrye/smallrye-jwt/blob/f7c76892723bce038bd82b7f0165e68a11c706f1/implementation/jwt-auth/src/main/java/io/smallrye/jwt/auth/principal/AbstractKeyLocationResolver.java#L88-L93

Reply:
@grdryn `typ` [JWT Spec](https://datatracker.ietf.org/doc/html/rfc7519#section-5.1) shorthand for type.